src/main/java/graphql/schema/GraphQLTypeCodeVisitor.java

```diff
@@ -0,0 +1,51 @@
+package graphql.schema;
+
+import graphql.Internal;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+
+import static graphql.Assert.assertTrue;
+import static graphql.util.TraversalControl.CONTINUE;
+
+/**
+ * This ensure that all fields have data fetchers and that unions and interfaces have type resolvers
+ */
+@Internal
+class GraphQLTypeCodeVisitor extends GraphQLTypeVisitorStub {
```

Comment:
make the class public. It is marked as Internal, so everybody can choose.

Reply:
also: I don't think the name GraphQLTypeCodeVisitor is so good. One reason is the whole GraphQLType thing, but we can tackle that when we do https://github.com/graphql-java/graphql-java/issues/1251 .

The other is: just name it maybe like CodeRegistryBuilder or so to better reflect what it does.

Reply:
> make the class public. It is marked as Internal, so everybody can choose.

But it does a specific job.  That is a visitor that finds all the "code" spots and transfers them to the code registry

In a later version when we stop people putting the "code in the types" then this class can die OR at least become a checker (that you HAVE a type resolver for each type).  Again a job that is internal and not for every one.

Reply:
as for naming this code - sure - what about CodeRegistryVisitor - since this is what it does.  Visits types to find "code" in them and build the code registry